hydromt_wflow/wflow.py

```diff
@@ -392,96 +424,230 @@ def setup_rivers(
         self.staticgeoms.pop("rivers", None)  # remove old rivers if in staticgeoms
         self.rivers  # add new rivers to staticgeoms
 
-    def setup_hydrodem(
+        # Add hydrologically conditioned elevation map for the river, if required
+        if river_routing == "local-inertial":
+            postfix = {"wflow_dem": "_avg", "dem_subgrid": "_subgrid"}.get(
+                elevtn_map, ""
+            )
+            name = f"hydrodem{postfix}"
+
+            ds_out = flw.dem_adjust(
+                da_flwdir=self.staticmaps[self._MAPS["flwdir"]],
+                da_elevtn=self.staticmaps[elevtn_map],
+                da_rivmsk=self.staticmaps[self._MAPS["rivmsk"]],
+                flwdir=self.flwdir,
+                connectivity=connectivity,
+                river_d8=True,
+                logger=self.logger,
+            ).rename(name)
+            self.set_staticmaps(ds_out)
+
+            # update toml model.river_routing
+            self.logger.debug(
+                f'Update wflow config model.river_routing="{river_routing}"'
+            )
+            self.set_config("model.river_routing", river_routing)
+
+            self.set_config("input.lateral.river.bankfull_depth", self._MAPS["rivdph"])
+            self.set_config("input.lateral.river.bankfull_elevation", name)
+        else:
+            self.set_config("model.river_routing", river_routing)
+
+    def setup_floodplains(
         self,
-        elevtn_map="wflow_dem",
-        river_routing="local-inertial",
-        land_routing="kinematic-wave",
+        hydrography_fn,
+        floodplain_type: str,
+        ### Options for 1D floodplains
+        river_upa: Optional[float] = None,
+        flood_depths: List = [0.5, 1.0, 1.5, 2.0, 2.5],
```

Comment:
is a max of 2.5m sufficient in most cases? I can image we need more layers.

Reply:
Good point! I have added values up until 5.0m